src/SFA.DAS.ApprenticePortal.UnitTests/Application/Homepage/Queries/GetApprenticeHomepageQueryTests.cs

```diff
@@ -0,0 +1,116 @@
+            Assert.AreEqual(result.apprenticeHomepage.apprentice.ApprenticeId, apprenticeId);
+            Assert.AreEqual(result.apprenticeHomepage.apprentice.FirstName, firstName);
+            Assert.AreEqual(result.apprenticeHomepage.apprentice.LastName, lastName);
+            Assert.AreEqual(result.apprenticeHomepage.apprenticeship.CourseName, courseName);
+            Assert.AreEqual(result.apprenticeHomepage.apprenticeship.EmployerName, employerName);
+        }
+
+        [Test, MoqAutoData]
+        public async Task TestGetApprenticeHomepageQueryNoApprenticeships(
+            [Frozen] Mock<IApprenticeAccountsApiClient<ApprenticeAccountsApiConfiguration>> accountsApiClient,
+            [Frozen] Mock<IApprenticeCommitmentsApiClient<ApprenticeCommitmentsApiConfiguration>> commitmentsApiClient,
+            GetApprenticeHomepageQueryHandler sut
+            )
+        {
+            //Arrange
+            SetupMoqApprentice();
+
+            accountsApiClient.Setup(client =>
+                client.Get<Apprentice>(It.IsAny<GetApprenticeRequest>()))
+                .ReturnsAsync(_moqApprentice);
+
+            SetupBlankMoqApprenticeships();
+
+            commitmentsApiClient.Setup(client =>
+                client.Get<GetApprenticeApprenticeshipsResult>(It.IsAny<GetApprenticeApprenticeshipsRequest>()))
+                .ReturnsAsync(new GetApprenticeApprenticeshipsResult { apprenticeships = _moqApprenticeships });
+
+            // Act
+            var result = await sut.Handle(new GetApprenticeHomepageQuery() { ApprenticeId = apprenticeId }, CancellationToken.None);
+
+            // Assert
+            Assert.NotNull(result.apprenticeHomepage);
+            Assert.AreEqual(result.apprenticeHomepage.apprentice.ApprenticeId, apprenticeId);
+            Assert.AreEqual(result.apprenticeHomepage.apprentice.FirstName, firstName);
+            Assert.AreEqual(result.apprenticeHomepage.apprentice.LastName, lastName);
+            Assert.IsNull(result.apprenticeHomepage.apprenticeship);            
+        }
+
+        private void SetupMoqApprentice()
+        {
+            _moqApprentice = _fixture.Build<Apprentice>()
+                .With(p => p.ApprenticeId, apprenticeId)
+                .With(p => p.FirstName, firstName)
+                .With(p => p.LastName, lastName)
+                .Create();
+        }
+
+        private void SetupMoqApprenticeships()
+        {
+            _moqApprenticeships = new List<Apprenticeship>() {
+                _fixture.Build<Apprenticeship>()
+                    .With(p => p.ApprenticeId, apprenticeId)
+                    .With(p => p.CourseName, courseName)
+                    .With(p => p.EmployerName, employerName)
+                    .With(p => p.ConfirmedOn, DateTime.Now)
+                    .Create()};
+        }
+
+        private void SetupBlankMoqApprenticeships()
+        {
+            _moqApprenticeships = new List<Apprenticeship>();                
+        }
+    }
+}
```

Comment:
A matter worth discussing, I missed it earlier. Is it better to test this code using Wiremock or using this traditional method.   

I'm not totally convinced client.Get<> is working the way we think it is and wiremock to me seems a much better testing strategy, as it's really testing the boundaries of the system and not making assumptions about the internal components. Ben may have an opinion on this.

 Wiremock also allows us to update the internals without having to refactor our tests